Set up branching in 2_process

amart90 opened the pull request:
### What changed
In this PR, the primary objective was to branch the `map_data` and `chart_data` targets across years. More comprehensively, in this PR I:
- Added the ability to load huc data as an ESRI geodatabase (the natively downloaded format)
- Added `.empty` to `1_fetch/in/` to establish file structure
- Noted data source for F2F2 data in `_targets.R`
- Split up target creation into `1_fetch.R`, `2_process.R`, and `3_visualize.R`
- Dynamically branched `chart_data` target over years (1984, 1985, ..., 2020)
- Dynamically branched `map_data` target over half-years (1984, 1984.5, ..., 2019.5, 2020)
- Attempted to branch `map_animation_png` over half years

### Review request
Cee, can you please check that the pipeline runs and that the branching is correct. I think 3_visualize is weird, so if you only do the following pipeline, it might save some errors or other weirdness:
```
tar_make(map_data)
tar_make(chart_animation_gif)
```

Note: I changed the HUC input code (`1_fetch.R`) so that it matches the data directly downloaded from the Box site. You can still use the shapefile I sent you as the input if you change the code in line 19 of `1_fetch.R` to read:
``` 
get_huc(file_in = ""1_fetch/in/F2F2_HUC12.shp",
              crs = 9311)),
```

ceenell commented on watershed-fire/2_process.R:
```diff
@@ -0,0 +1,30 @@
+p2_targets <- list(
+  
+  # Define years as target (so they can be mapped for the map png branching)
+  tar_target(Years,
+             seq(years[1], years[2])),
+  
+  tar_target(Years_0.5,
+             seq(years[1], years[2], by = 0.5)),
```
I like what you're doing here. I would suggest a slight modification to the naming of `Years` and `years` to avoid confusion, possibly changing `years` to `year_range` or something like that. Instead of using the indexing, you could also use `min(year_range)` and `max(year_range)` to make it a bit more flexible.

ceenell commented on watershed-fire/1_fetch.R:
```diff
@@ -0,0 +1,27 @@
+  
+  # To read a geodatabase, file_in = [path and name of geodatabase] and layer = [name of layer]
+  # To read a shapefile, file_in = [path and name shapefile]
+  tar_target(huc,
```
here you could use a slightl ymroe informative target name, like `f2f2_huc12` to correspond to the layer you're using

ceenell commented on watershed-fire/2_process/src/map_data.R:
```diff
@@ -1,9 +1,25 @@
-}
\ No newline at end of file
+    bind_cols(st_set_geometry(out, NULL)) %>%
+    add_row(Year = years + 0.5)
```
What is this doing?

amart90 replied:
Line 22 is binding the data from the polygon dataset to the centroid dataset.

Line 23 is add blank rows between years. To produce the embers glowing effect of the map, it has to fade into a blank map (or more specifically, into a map with subtle glowing of all the previous fires' "scars") between the annual glowpoint frames. This adds these "blank frames" to fade in and out of between years. Here is an example of what those frames look like.

2003
![image](https://user-images.githubusercontent.com/25806255/180872847-eadacf7c-ea47-4f20-89e3-bda7064fb165.png)

2003.5
![image](https://user-images.githubusercontent.com/25806255/180872885-9be47ac3-846b-4874-a30b-579769bb98f6.png)

2004
![image](https://user-images.githubusercontent.com/25806255/180873008-c2392fa3-6991-4ee0-a581-1f6661f6cc94.png)

Without these intermediate frames it gets difficult to discern what is happening each year.

ceenell replied:
smart!

ceenell reviewed (commented):
Nice job setting up the dynamic branching and addressing the comments in https://github.com/USGS-VIZLAB/idea-blitz/issues/15. I was able to build these targets, but first needed to add `3_visualize/out` for the gif to go to. Good move on adding the geodatabase step into the pipeline. I wouldn't worry about not being able to get it via R right now, your comments make the source clear.

My output:
![chart_animation](https://user-images.githubusercontent.com/17803537/180320672-637b9746-1921-48b3-ae96-018256b0927a.gif)
🥳  so cool

I've left a few comments below to address before merging, plus adding a `.empty` file in `3_visualize/out` so it gets committed even though it's empty.

amart90 commented:
@ceenell, thanks for the feedback. In this re-request I have addressed the following:
- renamed the target `huc` to the more descriptive `f2f2_huc12`
- changed the parameter `year` to the more descriptive `year_range`
- Added interpolated data (at the 0.5 year) to the `chart data` target (I called it "shart_data" in the commit message, lol, oops)
- combined the building of both charts into a single target (branched over years)

I got rid of the year text in the map, but I will add it back on the next PR - I want to mess with the styling a bit.

ceenell commented:
This looks really good, everything ran as expected.
![animation_2000](https://user-images.githubusercontent.com/17803537/181362847-52ffa847-1787-4210-8f24-805253fc7cf3.png)

The output pngs are ~1.3MB which might make the final gif pretty big, but we can deal with that when we get there. Go ahead and merge this and let's work on animating it.
